compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5694,8 +5691,7 @@ public void visit(BLangForkJoin forkJoin) {
 
     @Override
     public void visit(BLangLiteral literalExpr) {
-        if (literalExpr.getBType().tag == TypeTags.ARRAY
-                && ((BArrayType) literalExpr.getBType()).eType.tag == TypeTags.BYTE) {
+        if (literalExpr.getBType().tag == TypeTags.ARRAY) {
```

Comment:
With https://github.com/ballerina-platform/ballerina-lang/pull/37307, literalExpr.getBType() can be a typeReference. 
```suggestion
        if (Types.getReferredType(literalExpr.getBType()).tag == TypeTags.ARRAY) {
```

Reply:
Is this applicable for this https://github.com/ballerina-platform/ballerina-lang/blob/97e8d52b62113bda3abb212078e222f001bea4f3/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L869 as well ?